src/System.Buffers/src/System/Buffers/ArrayPoolEventSource.cs

```diff
@@ -0,0 +1,31 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Diagnostics.Tracing;
+
+namespace System.Buffers
+{
+    public sealed class ArrayPoolEventSource : EventSource
+    {
+        public ArrayPoolEventSource() : base() { }
+
+        [Event(1, Message = "Rented Buffer {0} of size {1} from Bucket {2}", Level = EventLevel.Informational)]
```

Comment:
Hmm, what's the run-time cost of localizing these? Unlike exception messages, these need to be very efficient. @vancem?